analytics/terraform/datahub-on-eks/datahub-addon/modules/prereq/main.tf

```diff
@@ -31,8 +31,13 @@ resource "random_password" "master_password" {
   min_special = 1
 }
 
-resource "aws_opensearch_domain" "es" {
+resource "time_sleep" "wait_10_seconds" {
   depends_on     = [aws_iam_service_linked_role.es]
+  create_duration = "10s"
+}
+
+resource "aws_opensearch_domain" "es" {
+  depends_on     = [time_sleep.wait_10_seconds]
```

Comment:
@bryantbiggs for advice. Is there a better way?

Reply:
service linked roles should be treated as a prerequisite and I wouldn't encourage creating those in code

If someone has created an opensearch domain through the console, this code will fail because the SLR already exists. For example, in [Karpenter's getting started](https://karpenter.sh/docs/getting-started/getting-started-with-karpenter/) we put:

>aws iam create-service-linked-role --aws-service-name spot.amazonaws.com || true
> // If the role has already been successfully created, you will see:
> // An error occurred (InvalidInput) when calling the CreateServiceLinkedRole operation: Service role name AWSServiceRoleForEC2Spot has been taken in this account, please try a different suffix.

Reply:
Creating SLR is optional - there is a variable `create_iam_service_linked_role_es` controlling it.   This fix is needed when the customer doesn't have SLR and opt to have it created by the blueprint.   Previously the opensearch domain depended on the SLR directly, and deployment often failed as IAM is eventual consistent.   Adding the delay of a few seconds between the two resources worked well in testing.

Reply:
Optional or not, its rather precarious and these are patterns, not modules. There are several SLRs used here, we don't codify those so I would say we should not codify this one either and simply state its a prerequisite and give the CLI one-liner like shown above for the spot SLR

Reply:
@bbgu1 Could you please update as recommended by @bryantbiggs so that we can merge this PR.